### PR TITLE
Testing REST API integrations using WireMock and MockServer

### PR DESCRIPTION
**Pull request by sdelamo**
Port of Spring Boot Test Containers guide to Micronaut: https://testcontainers.com/guides/testing-rest-api-integrations-using-wiremock/

To generate guide: 

```
./gradlew testingRestApiIntegrationsUsingMockserverBuild
```

PDF: 
[Testing REST API integrations using WireMock.pdf](https://github.com/micronaut-projects/micronaut-guides/files/12608714/Testing.REST.API.integrations.using.WireMock.pdf)

### CODE REVIEW

File: guides/testing-rest-api-integrations-using-mockserver/testing-rest-api-integrations-using-mockserver.adoc [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+common:requirements.adoc[]
+* https://www.testcontainers.org/supported_docker_environment/[A Docker environment supported by Testcontainers]
+
+common:completesolution.adoc[]
+
+== What we are going to achieve in this guide
+We will create a Micronaut application that talks to an external REST API.
+Then, we will test the external REST API integration using
+the https://testcontainers.com/modules/wiremock/[Testcontainers WireMock module].
```
Comment (sivaprasadreddy) [inline guides/testing-rest-api-integrations-using-mockserver/testing-rest-api-integrations-using-mockserver.adoc]: This guide explains how to test using both **WireMock** and **MockServer**. This sentence needs to be updated to reflect that.

File: guides/testing-rest-api-integrations-using-mockserver/testing-rest-api-integrations-using-mockserver.adoc [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+common:create-app-features.adoc[]
+
+== WireMock dependency
+
+Once the application is generated, add the *WireMock* library as a test dependency.
+
+dependency:mockserver[groupId=org.testcontainers,scope=test]
```
Comment (sivaprasadreddy) [inline guides/testing-rest-api-integrations-using-mockserver/testing-rest-api-integrations-using-mockserver.adoc]: This is MockServer dependency. This should be changed to WireMock dependency.

File: guides/testing-rest-api-integrations-using-mockserver/testing-rest-api-integrations-using-mockserver.adoc [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+https://www.mock-server.com/[MockServer]
+
+____
+For any system you integrate with via HTTP or HTTPS MockServer can be used as a mock configured to return specific responses for different requests, a proxy recording and optionally modifying requests and responses, both a proxy for some requests and a mock for other requests at the same time.
+____
+
+Add the MockServer netty dependency:
+
+dependency:mockserver-netty[groupId=org.mock-server,scope=test,version=@mockserver-nettyVersion@]]
```
Comment (sivaprasadreddy) [inline guides/testing-rest-api-integrations-using-mockserver/testing-rest-api-integrations-using-mockserver.adoc:189]: Here we need to add testcontainers mockserver dependency.

File: guides/testing-rest-api-integrations-using-mockserver/testing-rest-api-integrations-using-mockserver.adoc (hunk 1 above)
Comment (sdelamo) [reply guides/testing-rest-api-integrations-using-mockserver/testing-rest-api-integrations-using-mockserver.adoc]: resolved via https://github.com/micronaut-projects/micronaut-guides/pull/1330/commits/c8926945aac1a1269f545be187bc65e114c5ef2e

File: guides/testing-rest-api-integrations-using-mockserver/testing-rest-api-integrations-using-mockserver.adoc (hunk 3 above)
Comment (sdelamo) [reply guides/testing-rest-api-integrations-using-mockserver/testing-rest-api-integrations-using-mockserver.adoc:189]: resolved via https://github.com/micronaut-projects/micronaut-guides/pull/1330/commits/296eb6593a7dd3dc8cd492f0a0869f448e5d7e39

File: guides/testing-rest-api-integrations-using-mockserver/testing-rest-api-integrations-using-mockserver.adoc (hunk 2 above)
Comment (sdelamo) [reply guides/testing-rest-api-integrations-using-mockserver/testing-rest-api-integrations-using-mockserver.adoc]: resolved via https://github.com/micronaut-projects/micronaut-guides/pull/1330/commits/296eb6593a7dd3dc8cd492f0a0869f448e5d7e39

**Review by timyates — APPROVED** [review]

**Review by sivaprasadreddy — APPROVED** [review]

**Pull request merged**